src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -978,16 +977,223 @@ void CmdSketcherViewSection::activated(int iMsg)
 
 bool CmdSketcherViewSection::isActive()
 {
-    Gui::Document *doc = getActiveGuiDocument();
+    return isEditModeToolActive(getActiveGuiDocument());
+}
+
+/* Grid tool */
+class GridSpaceAction : public QWidgetAction
+{
+public:
+    GridSpaceAction(QObject* parent) : QWidgetAction(parent) {
+    }
+
+protected:
+    QWidget* createWidget(QWidget* parent) override
+    {
+        auto* sizeLabel = new QLabel(QString::fromLatin1("Size"));
+        auto* gridSizeBox = new Gui::QuantitySpinBox();
+        gridSizeBox->setProperty("unit", QVariant(QString::fromLatin1("mm")));
+        gridSizeBox->setObjectName(QString::fromLatin1("gridSize"));
+        gridSizeBox->setToolTip(QApplication::translate("Sketcher_GridSize", "Distance between two subsequent grid lines"));
+        //Is General/GridSize correct? From taskSketcherGeneral it seems yes.
+        ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General/GridSize");
+        gridSizeBox->setValue(hGrp->GetFloat("GridSize", 10.0));
+        gridSizeBox->setSingleStep(1.0);
+        gridSizeBox->setMaximum(99999999.0);
+        gridSizeBox->setMinimum(0.001);
+        gridSizeBox->setDecimals(3);

```

Comment:
For the number of decimals? 
Do you know the pref path and name?

Reply:
That's an open question to me. ;) Maybe @wwmayer can advise on this.
I don't know by heart how to get it, but probably pretty easy to find (for example look into QuantitySpinBox class). ;)

Reply:
Found it, fiy: 

```
    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath
        ("User parameter:BaseApp/Preferences/Units");
    hGrp->SetInt("Decimals", ui->spinBoxDecimals->value());
```

Reply:
Use `Base::UnitsAPI::getDecimals`

Reply:
I think that if the value isn't set at all, the QuantitySpinBox will apply user setting. Am I correct ?

Reply:
This would make sens. So if it was set to 3 in the first place, it's probably the original designer of the gridsize who thought it best to override the default by 3. Not sure it's best though, I would rather use user pref no?

Reply:
I vote for user pref too (just my opinion). This seems sensible.

Reply:
Base::UnitsAPI::getDecimals() caches user settings. Its internal value is set at program start according to user config and when the user changes the settings the value will be adjusted.